
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="generator" content="Docutils 0.17: http://docutils.sourceforge.net/" />

    <title>Change log &#8212; TSOPPI: TSO500 Post-Processing for InPreD v0.3.0:22-02-28 documentation</title>
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="_static/alabaster.css" type="text/css" />
    <script id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
    <script src="_static/jquery.js"></script>
    <script src="_static/underscore.js"></script>
    <script src="_static/doctools.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="prev" title="Frequently asked questions" href="FAQ.html" />
   
  <link rel="stylesheet" href="_static/custom.css" type="text/css" />
  
  
  <meta name="viewport" content="width=device-width, initial-scale=0.9, maximum-scale=0.9" />

  </head><body>
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          

          <div class="body" role="main">
            
  <section id="change-log">
<h1>Change log<a class="headerlink" href="#change-log" title="Permalink to this headline">¶</a></h1>
<p>22-03-20 (TSOPPI version 0.3.1)</p>
<ul class="simple">
<li><p>bug fix:</p>
<ul>
<li><p>in the tumor content adjusted plots within the *_CNV_distribution_plots.pdf
output file, patient sample values below the visible range were plotted at
the maximum visible values instead of being plotted at the minimum visible values;
this bug is now fixed.</p></li>
</ul>
</li>
</ul>
<p>22-03-10 (TSOPPI version 0.3.0)</p>
<ul class="simple">
<li><p>updated:</p>
<ul>
<li><p>continuumio/miniconda3 base Docker image of TSOPPI updated to version 4.10.3p1;</p></li>
<li><p>PCGR and CPSR updated to version 1.0.0.</p></li>
</ul>
</li>
<li><p>added:</p>
<ul>
<li><p>gene-wise CNV distribution plots based on a set of 85 normal InPreD samples;</p></li>
<li><p>inclusion of LocalApp’s Nirvana variant annotation (allowing for recognition of coding variants according to RefSeq,
despite their “intergenic” PCGR annotation);</p></li>
<li><p>enabling use of InPreD’s own small variant blacklist intended for variants
appearing in the variant interpretation table of most samples;
variants present on this blacklist are being redirected into a new file (*small_variant_table_blacklisted.tsv);</p></li>
<li><p>enabling use of InPreD’s own gene- and small variant- whitelists intended
for treatment-relevant sites/regions; details regarding these sites/regions
are being saved into a new file (*whitelist_details.tsv);</p></li>
<li><p>new output files related to DNA coverage (*coverage_plots.pdf,
*coverage_histogram.tsv, *coverage_details.tsv.gz) are being created for DNA tumor samples;</p></li>
<li><p>jdk-11.0.6 and picard_2.26.2 (namely its tools &quot;AddOrReplaceReadGroups&quot; and &quot;CollectSequencingArtifactMetrics&quot;)
can now be utilized in order to calculate sequencing artifact metrics for the tumor DNA sample
(option &quot;–enable_CSAM_check True/False&quot;);</p></li>
<li><p>patient sex is now taken into account in CNV processing and plotting:</p>
<ul>
<li><p>chrX non-PAR genes are assumed to have 1 copy in copy number neutral male patients;</p></li>
<li><p>LocalApp-assumed patient sex (derived from tumor DNA sample data) is now being mentioned in more output files;</p></li>
</ul>
</li>
<li><p>PCGR’s mutational signature analysis is being automatically run for samples with &gt;=100 assumed somatic SNVs;</p></li>
<li><p>on-screen logs of the main sample-processing TSOPPI shell script (&quot;process_patient_samples.sh&quot;)
are now being saved into dedicated files;</p></li>
<li><p>a new file integrating splicing variant and fusion variant output meant for manual QC and review
is now being generated (*fusion_and_splice_variant_candidates.tsv);
the new table is based on LocalApp’s final fusion output and includes
events generated by both Manta/RnaFusionFilter and SpliceGirl;</p></li>
<li><p>BL variants are now being annotated by CPSR (they used to be ignored in the CPSR processing previously,
which was inconsistent with the PCGR annotation).</p></li>
</ul>
</li>
<li><p>changed:</p>
<ul>
<li><p><a class="reference internal" href="metrics_plotting.html"><span class="doc">the metrics plotting tool</span></a> now has to be invoked via a wrapper bash script
(“process_metrics_files.sh”) that first activates the necessary Conda environment
(the parameter setup has not been changed however);</p></li>
<li><p>“–create_plots” {True,False} option has been added to the metrics plotting tool; disabling
the metrics plotting functionality can be advantageous when the only desired
output are text-based metrics tables (typically in scenarios when many
runs are being processed together);</p></li>
<li><p>parameter “–run_completion_status_file” of the metrics plotting tool can
now have value &quot;NA&quot; (if for example a suitable RunCompletionStatus.xml
file is not available);</p></li>
<li><p>WARNING messages are now being output to STDERR logs;</p></li>
<li><p>multiple changes to the CNV summary output tables:</p>
<ul>
<li><p>*merged_CNV_summary_FC_sorted.tsv (replacing *merged_CNV_summary_CN_sorted.tsv) is now being sorted according to each gene’s LocalApp-reported FC value (fold change);</p></li>
<li><p>*merged_CNV_summary_location_sorted.tsv is now being sorted according to genomic position with the expected chromosome order;</p></li>
<li><p>&quot;Expected_germline_CN&quot; column has been added to both above-mentioned tables;</p></li>
</ul>
</li>
<li><p>output of the &quot;summarize_run_variants.py&quot; tool is now sorted by sample InPreD IDs;</p></li>
<li><p>the &quot;LoH&quot; entry from LocalApp’s CNV results is no longer presented as one of the targeted genes
in gene-wise tables and plots generated during the post-processing;</p></li>
<li><p>the RNA and DNA post-processing tools have been merged,
many changes have been made to the resulting tool (&quot;process_patient_samples.sh&quot;):</p>
<ul>
<li><p>supplying a reference FASTA file is now mandatory;</p></li>
<li><p>short options have been removed;</p></li>
<li><p>multiple parameter names have been changed;</p></li>
<li><p>keeping/deleting intermediate files is now optional (option &quot;–remove_temporary_files True/False&quot;);</p></li>
<li><p>if matched samples (normal DNA and/or tumor RNA) are provided on input together with a tumor DNA sample,
concordance values based on known germline SNVs are calculated and reported
in a new output file (*_sample_concordance.tsv);</p></li>
<li><p>a file listing details about all input samples (sample_list.tsv) is being created in the output directory;</p></li>
<li><p>if a matched tumor RNA sample is provided on input, small variants found
in the tumor DNA sample will be investigated in the tumor RNA sample
using bcftools mpileup (the RNA small variant information will be reflected in e.g.,
the IGV links documents and the small variant interpretation table);</p></li>
<li><p>a joint mutational signature plot is now being automatically created for each DNA tumor samples
(this is just a plot, which shouldn’t be confused with full signature mutational analysis);</p></li>
</ul>
</li>
<li><p>changes to the variant interpretation table:</p>
<ul>
<li><p>“MUTATION_HOTSPOT” field from PCGR/CPSR has been added;</p></li>
<li><p>a new field detailing Foundation One Liquid CDx targets has been added (“F1LCDx_targets”);</p></li>
<li><p>RNA sample fields (“Depth_tumor_RNA” and “AF_tumor_RNA”) have been added;</p></li>
<li><p>values necessary for some of the filters to trigger were lowered
(“LOW_TUMOR_DP”: 50 -&gt; 20; “LOW_TUMOR_VAF”: 0.05 -&gt; 0.02);</p></li>
<li><p>fields “Class_judgement” and “Class_judgement_comments” are now
generated pre-filled with the default values “include” and “.”, unless
conditions for automated variant exclusion are being met (please refer to the table’s header for details);</p></li>
<li><p>the order in which the included variants are listed has been changed:</p>
<ul>
<li><p>recognized hotspot variants (i.e., variants with a non-NA entry in
the ‘Mutation_hotspot’ field) are now listed at the top,
variants with an “NA” entry in the ‘Mutation_hotspot’ field are listed afterwards;</p></li>
<li><p>within each of the two groups mentioned above,
the variants are further sorted first by Tier (alphabetically),
then by Coding status (alphabetically),
and finally by TSO500 class (reverse-alphabetically);</p></li>
</ul>
</li>
<li><p>inclusion criteria based on protein change consequences (as reported by Nirvana) have been extended
(the following SequenceOntology terms now trigger inclusion: frameshift_variant,
start_lost, stop_lost, stop_gained, splice_acceptor_variant, splice_donor_variant)</p></li>
<li><p>some gene names have been harmonized in order to increase compatibility between
the LocalApp and PCGR/CPSR (&quot;MYCL1&quot; -&gt; &quot;MYCL&quot;, &quot;C11orf30&quot; -&gt; &quot;C11ORF30&quot;);</p></li>
<li><p>“Coding_status” field format changes:</p>
<ul>
<li><p>the “_variant” affix in the standard SequenceOntology terms is no longer being removed;</p></li>
<li><p>status “x_noncoding” is now reported as “x:noncoding_variant” instead;</p></li>
</ul>
</li>
<li><p>field names changes in the VIT;</p></li>
</ul>
</li>
<li><p>changes to TSOPPI’s plots:</p>
<ul>
<li><p>the “[deprecated]” keyword has been removed from sample QC plots;</p></li>
<li><p>Illumina’s callability metric (percentage of exon bases with coverage &gt;= 50)
has been added to DNA sample QC plots;</p></li>
<li><p>contamination run metrics plots have been slightly adjusted for visual clarity;</p></li>
<li><p>multiple changes to CNV plots:</p>
<ul>
<li><p>adding detailed chromosome-wise VAF plots with labels for variants included in the interpretation table;</p></li>
<li><p>adding chromosome-wise gene CNV plots;</p></li>
<li><p>adding a genome-wide small variant VAF plot with variant sequential order on the x-axis (instead of genomic location);</p></li>
</ul>
</li>
</ul>
</li>
<li><p>row names/numbers are no longer present in the intermediate “master_metrics_table.tsv” file produced during run metrics plotting
(the column/field headers now refer to the correct data items);</p></li>
<li><p>headers are now present in both small variant overview tables.</p></li>
</ul>
</li>
<li><p>removed:</p>
<ul>
<li><p>IGV snapshot creation functionality.</p></li>
</ul>
</li>
<li><p>caveats:</p>
<ul>
<li><p>the patient sex, as estimated by the LocalApp, can be wrong.</p></li>
</ul>
</li>
</ul>
<dl class="simple">
<dt>21-06-07 (TSOPPI version 0.1)</dt><dd><ul class="simple">
<li><p>fixing broken IGV port command links.</p></li>
</ul>
</dd>
<dt>21-06-02</dt><dd><ul class="simple">
<li><p>harmonization of parameter nomenclature across all TSOPPI tools
(please note: this implies numerous parameter name changes in the tool set);</p></li>
<li><p>introduction of a new parameter to the DNA and RNA post-processing tools:
“–inpred_nomenclature” (<a class="reference internal" href="inpred_nomenclature.html"><span class="doc">InPreD sample ID nomenclature</span></a>
will be assumed to be in use only if this parameter is set to “True”);</p></li>
<li><p>when applicable, the new InPreD sample ID nomenclature is now reflected
in all sample-wise QC plots;</p></li>
<li><p>changing multiple internal parameter values in the
<cite>DNA sample post-processing tool</cite> [deprecated as of v.0.3]
(these parameters don’t affect which variants will be present in the output files,
they only affect how the output variants will be flagged):
MIN_TUMOR_DP: 10 -&gt; 50; MIN_TUMOR_VAF: 0.03 -&gt; 0.05; MAX_TUMOR_VAF: 0.98 -&gt; 0.99;</p></li>
<li><p>changing “*htm” files into “*html” files.</p></li>
</ul>
</dd>
<dt>21-05-24</dt><dd><ul class="simple">
<li><p>when utilizing a normal sample, the pipeline version string should now
correctly conveys that information (stating “TN”, instead of the previous erroneous “T”);</p></li>
<li><p>genome-wide CNV plots now display centromeres,
BAF plots now show GL_P variants;</p></li>
<li><p>instead of the number of processed samples, the variant recurrence strings
now show the number of callable samples for given variant position.</p></li>
</ul>
</dd>
<dt>21-04-14</dt><dd><ul class="simple">
<li><p>initial version.</p></li>
</ul>
</dd>
</dl>
<p>(last updated: 2022-03-04)</p>
</section>


          </div>
          
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
<h1 class="logo"><a href="index.html">TSOPPI: TSO500 Post-Processing for InPreD</a></h1>








<h3>Navigation</h3>
<p><span class="caption-text">Table of contents:</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="getting_started.html">Getting started</a></li>
<li class="toctree-l1"><a class="reference internal" href="metrics_plotting.html"><strong>Metrics plotting</strong> tool</a></li>
<li class="toctree-l1"><a class="reference internal" href="variant_summary_creation.html"><strong>Variant summary creation</strong> tool</a></li>
<li class="toctree-l1"><a class="reference internal" href="variant_recurrence_table_creation.html"><strong>Variant recurrence table creation</strong> tool</a></li>
<li class="toctree-l1"><a class="reference internal" href="sample_data_postprocessing.html"><strong>Sample data post-processing</strong> tool</a></li>
<li class="toctree-l1"><a class="reference internal" href="inpred_nomenclature.html">InPreD sample ID nomenclature</a></li>
<li class="toctree-l1"><a class="reference internal" href="software_versions.html">Software versions</a></li>
<li class="toctree-l1"><a class="reference internal" href="FAQ.html">Frequently asked questions</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Change log</a></li>
</ul>

<div class="relations">
<h3>Related Topics</h3>
<ul>
  <li><a href="index.html">Documentation overview</a><ul>
      <li>Previous: <a href="FAQ.html" title="previous chapter">Frequently asked questions</a></li>
  </ul></li>
</ul>
</div>
<div id="searchbox" style="display: none" role="search">
  <h3 id="searchlabel">Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" />
      <input type="submit" value="Go" />
    </form>
    </div>
</div>
<script>$('#searchbox').show(0);</script>








        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="footer">
      &copy;2022, Daniel Vodak.
      
      |
      Powered by <a href="http://sphinx-doc.org/">Sphinx 3.4.3</a>
      &amp; <a href="https://github.com/bitprophet/alabaster">Alabaster 0.7.12</a>
      
      |
      <a href="_sources/changelog.rst.txt"
          rel="nofollow">Page source</a>
    </div>

    

    
  </body>
</html>