
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="generator" content="Docutils 0.17: http://docutils.sourceforge.net/" />

    <title>Variant recurrence table creation tool &#8212; TSOPPI: TSO500 Post-Processing for InPreD v0.3.0:22-02-28 documentation</title>
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="_static/alabaster.css" type="text/css" />
    <script id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
    <script src="_static/jquery.js"></script>
    <script src="_static/underscore.js"></script>
    <script src="_static/doctools.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Sample data post-processing tool" href="sample_data_postprocessing.html" />
    <link rel="prev" title="Variant summary creation tool" href="variant_summary_creation.html" />
   
  <link rel="stylesheet" href="_static/custom.css" type="text/css" />
  
  
  <meta name="viewport" content="width=device-width, initial-scale=0.9, maximum-scale=0.9" />

  </head><body>
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          

          <div class="body" role="main">
            
  <section id="variant-recurrence-table-creation-tool">
<h1><strong>Variant recurrence table creation</strong> tool<a class="headerlink" href="#variant-recurrence-table-creation-tool" title="Permalink to this headline">¶</a></h1>
<section id="about">
<h2>About<a class="headerlink" href="#about" title="Permalink to this headline">¶</a></h2>
<p>Small variant (SNV/short INDEL) quality control and interpretation is the most demanding part
of TSO500 results evaluation. Availability of matched control samples makes
this work considerably easier by greatly reducing the uncertainty of origin
(germline/somatic/artifact)
of individual variants found in the tumor sample. In the absence of matched control
samples, simple information about a variant’s recurrence in previously sequenced
samples can go a long way in assessing a variant’s origin.</p>
<p>TSOPPI’s “update_variant_recurrence_table” tool is meant to answer two types
of questions with regard to relevant (tumor and/or normal) samples investigated
so far in the InPreD project:</p>
<ul class="simple">
<li><p>How often is a given variant position usable for small variant calling? For
various reasons, variant calling might be difficult/impossible at certain loci.</p></li>
<li><p>How often does given small variant appear in tumor and/or normal InPreD samples?
The observed variant recurrence is broken down to several predefined variant
allelic fraction (VAF) tiers, which make it easier to judge a variant’s typical origin.</p></li>
</ul>
<p>The tool processes an arbitrary number of LocalApp-generated small variant VCF files
and outputs a table with recurrence information entry for each variant observed
in at least one sample (provided that the variant site was callable,
but regardless of how the variant has been classified).
The individual information entries can then be attached to the corresponding
variants’ output whenever these are encountered during sample post-processing
(please refer to the <a class="reference internal" href="sample_data_postprocessing.html"><span class="doc">sample data post-processing</span></a>
TSOPPI tool for details regarding its usage).</p>
<section id="additional-notes">
<h3>Additional notes<a class="headerlink" href="#additional-notes" title="Permalink to this headline">¶</a></h3>
<ul class="simple">
<li><p>The default read-depth callability threshold is set to 20.</p></li>
<li><p>Compared to single-nucleotide variants (SNVs), long insertions require their
supporting reads to cover a larger region base pair-wise - an SNV and an
insertion located at the same genomic position might therefore have different
callability within the same sample. However, the callability values recorded
in the variant recurrence tables produced by this tool are position-specific
(not variant-specific): If an input VCF includes at least one record that would
make a given position callable, the position will be considered callable for
all variants in the corresponding sample.</p></li>
<li><p>We recommend not using more than one tumor sample and one normal sample for any
given individual as input for this tool. The resulting recurrence information is
most useful if it is clear how many different individuals (rather than samples)
so far contained given variant.</p></li>
</ul>
</section>
</section>
<section id="input-files">
<h2>Input files<a class="headerlink" href="#input-files" title="Permalink to this headline">¶</a></h2>
<p>A single input file is required: A two-field table specifying the VCF
files that should be used for variant recurrence table construction.
The two expected fields are:</p>
<blockquote>
<div><dl class="field-list simple">
<dt class="field-odd">field 1</dt>
<dd class="field-odd"><p>an absolute path to a given VCF file</p>
</dd>
<dt class="field-even">field 2</dt>
<dd class="field-even"><p>a sample type code specifying whether given VCF represents
a tumor sample or a normal sample; the only recognized values in
this field are “T” (for tumors) and “N” (for normals)</p>
</dd>
</dl>
</div></blockquote>
<p>Any lines starting with the “#” symbol will be considered to be comments
(these lines will be ignored during the processing).
Here is an example of a VCF file table specifying 3 tumor sample VCF files
and 2 normal sample VCF files (additional comment-lines are also present):</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="c1"># VCF files for run_1</span>
<span class="o">/</span><span class="n">hs_prefix_path</span><span class="o">/</span><span class="n">analysis</span><span class="o">/</span><span class="n">run1</span><span class="o">/</span><span class="n">Results</span><span class="o">/</span><span class="n">pair_A</span><span class="o">/</span><span class="n">tumor_A</span><span class="o">/</span><span class="n">tumor_A_MergedSmallVariants</span><span class="o">.</span><span class="n">genome</span><span class="o">.</span><span class="n">vcf</span> <span class="n">T</span>
<span class="o">/</span><span class="n">hs_prefix_path</span><span class="o">/</span><span class="n">analysis</span><span class="o">/</span><span class="n">run1</span><span class="o">/</span><span class="n">Results</span><span class="o">/</span><span class="n">pair_A</span><span class="o">/</span><span class="n">normal_A</span><span class="o">/</span><span class="n">normal_A_MergedSmallVariants</span><span class="o">.</span><span class="n">genome</span><span class="o">.</span><span class="n">vcf</span> <span class="n">N</span>
<span class="o">/</span><span class="n">hs_prefix_path</span><span class="o">/</span><span class="n">analysis</span><span class="o">/</span><span class="n">run1</span><span class="o">/</span><span class="n">Results</span><span class="o">/</span><span class="n">pair_B</span><span class="o">/</span><span class="n">tumor_B</span><span class="o">/</span><span class="n">tumor_B_MergedSmallVariants</span><span class="o">.</span><span class="n">genome</span><span class="o">.</span><span class="n">vcf</span> <span class="n">T</span>
<span class="o">/</span><span class="n">hs_prefix_path</span><span class="o">/</span><span class="n">analysis</span><span class="o">/</span><span class="n">run1</span><span class="o">/</span><span class="n">Results</span><span class="o">/</span><span class="n">pair_B</span><span class="o">/</span><span class="n">normal_B</span><span class="o">/</span><span class="n">normal_B_MergedSmallVariants</span><span class="o">.</span><span class="n">genome</span><span class="o">.</span><span class="n">vcf</span> <span class="n">N</span>
<span class="o">/</span><span class="n">hs_prefix_path</span><span class="o">/</span><span class="n">analysis</span><span class="o">/</span><span class="n">run1</span><span class="o">/</span><span class="n">Results</span><span class="o">/</span><span class="n">pair_C</span><span class="o">/</span><span class="n">tumor_C</span><span class="o">/</span><span class="n">tumor_C_MergedSmallVariants</span><span class="o">.</span><span class="n">genome</span><span class="o">.</span><span class="n">vcf</span> <span class="n">T</span>
<span class="c1">#/hs_prefix_path/analysis/run1/Results/pair_C/normal_C/normal_C_MergedSmallVariants.genome.vcf</span>
</pre></div>
</div>
<section id="usage-notes">
<h3>Usage notes<a class="headerlink" href="#usage-notes" title="Permalink to this headline">¶</a></h3>
<ul class="simple">
<li><p>Header of the resulting variant recurrence table (VRT) keeps track of which VCF files
have been used during its construction (by storing the VCF file paths).
If the VRT output file specified with parameter
<em>--variant_recurrence_table</em> already exists, instead of simple replacement,
the tool will aim to update the VRT with variant information
from VCF files that aren’t yet listed in the VRT header (i.e., VCF files
already listed in the VRT header will be ignored during the update, while
information from novel VCF files will be merged with pre-existing VRT
variant entries).</p></li>
<li><p>For the purposes of VRT construction, please use VCF files with path
<strong>[LocalApp_run_output]/Results/[Pair_ID]/[sample_ID]/[sample_ID]_MergedSmallVariants.genome.vcf</strong>.</p></li>
</ul>
<p>Please consult the <a class="reference internal" href="#output-files-label"><span class="std std-ref">Output files</span></a> section for specification of the
VRT format.</p>
</section>
</section>
<section id="running-the-tool">
<h2>Running the tool<a class="headerlink" href="#running-the-tool" title="Permalink to this headline">¶</a></h2>
<p>Command line options:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>usage: update_variant_recurrence_table.py [-h] [-v] -i VCF_FILE_TABLE -o
                                        VARIANT_RECURRENCE_TABLE -s
                                        HOST_SYSTEM_MOUNTING_DIRECTORY
                                        [-c CONTAINER_MOUNTING_DIRECTORY]
                                        [-m MINIMUM_DEPTH]

Update given variant recurrence table with variants from samples listed in the
specified VCF file table.

  -i VCF_FILE_TABLE, --vcf_file_table VCF_FILE_TABLE
                        a table listing sample VCF files (as absolute paths,
                        in column 1) together with corresponding sample type
                        codes (in column 2; the only recognized codes are &quot;T&quot;
                        for tumors and &quot;N&quot; for normals); the supplied
                        parameter value should itself be an absolute path
  -o VARIANT_RECURRENCE_TABLE, --variant_recurrence_table VARIANT_RECURRENCE_TABLE
                        absolute path to the variant recurrence table that
                        should be created/updated with data retrieved from the
                        VCF files listed within the the supplied VCF file
                        table
  -s HOST_SYSTEM_MOUNTING_DIRECTORY, --host_system_mounting_directory HOST_SYSTEM_MOUNTING_DIRECTORY
                        absolute path to the host system mounting directory;
                        the specified directory should include all input and
                        output file paths in its directory tree
optional arguments:
  -h, --help            show this help message and exit
  -v, --version         show program`s version number and exit
  -c CONTAINER_MOUNTING_DIRECTORY, --container_mounting_directory CONTAINER_MOUNTING_DIRECTORY
                        container`s inner mounting point; the host system
                        mounting directory path/prefix will be replaced by the
                        container mounting directory path in all input and
                        output file paths (this parameter likely shouldn`t be
                        changed); default value: /inpred/data
  -m MINIMUM_DEPTH, --minimum_depth MINIMUM_DEPTH
                        minimum allowed read depth at the variant site; when
                        processing a given sample VCF, variants at genomic
                        sites with read coverage below this threshold will be
                        disregarded during the variant recurrence table
                        creation (the variant site might still have high-
                        enough coverage in other samples though);
                        default value: 20
</pre></div>
</div>
<p>Example invocation using the Docker image:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ [sudo] docker run \
    --rm \
    -it \
    -v /hs_prefix_path:/inpred/data \
    inpred/tsoppi_main:v0.1 \
      python /inpred/user_scripts/update_variant_recurrence_table.py \
        --vcf_file_table /hs_prefix_path/postprocessing/VCF_file_table_[date].tsv \
        --variant_recurrence_table /hs_prefix_path/postprocessing/variant_recurrence_table_[date].tsv \
        --host_system_mounting_directory /hs_prefix_path
</pre></div>
</div>
</section>
<section id="output-files">
<span id="output-files-label"></span><h2>Output files<a class="headerlink" href="#output-files" title="Permalink to this headline">¶</a></h2>
<p>A single output file is generated by this tool. Its header includes three types of
lines:</p>
<ul class="simple">
<li><p>lines with string “#[sample_vcf]” in the first field specify which samples
have been used during the table construction (this information can be utilized
during an update);</p></li>
<li><p>lines with a leading “#” symbol, but without the “#[sample_vcf]” value in the first field,
describe the format of the variant data lines in the table;</p></li>
<li><p>a line providing variant data field names:
“variant_id”, “tumor_recurrence_summary”, “normal_recurrence_summary”, “total_recurrence_summary”.</p></li>
</ul>
<p>The “variant_id” field values are in format “chromosome:position:ref_allale&gt;alt_allele”,
with positions being 1-based.</p>
<p>The summary string values within the variant data lines have format “X:A+B+C+D=M/N”, where:</p>
<ul class="simple">
<li><p><strong>X</strong> is the sample type, one of “T” = tumor, “N” = normal, “A” = any (the latter encompassing both tumor and normal samples);</p></li>
<li><p><strong>A</strong> is the number of samples of type X in which given variant was seen with VAF &lt; 0.01;</p></li>
<li><p><strong>B</strong> is the number of samples of type X in which given variant was seen with 0.01 &lt;= VAF &lt; 0.05;</p></li>
<li><p><strong>C</strong> is the number of samples of type X in which given variant was seen with 0.05 &lt;= VAF &lt; 0.35;</p></li>
<li><p><strong>D</strong> is the number of samples of type X in which given variant was seen with 0.35 &lt;= VAF;</p></li>
<li><p><strong>M</strong> is the number of samples of type X in which given variant was seen with any VAF;</p></li>
<li><p><strong>N</strong> is the number of investigated samples of type X in which the given variant’s site was callable (i.e., the variant site had coverage &gt;= 20).</p></li>
</ul>
<p>Values <strong>A</strong>, <strong>B</strong>, <strong>C</strong>, <strong>D</strong> and <strong>M</strong> also relate to callable samples only.</p>
<p>(last updated: 2022-03-03)</p>
</section>
</section>


          </div>
          
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
<h1 class="logo"><a href="index.html">TSOPPI: TSO500 Post-Processing for InPreD</a></h1>








<h3>Navigation</h3>
<p><span class="caption-text">Table of contents:</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="getting_started.html">Getting started</a></li>
<li class="toctree-l1"><a class="reference internal" href="metrics_plotting.html"><strong>Metrics plotting</strong> tool</a></li>
<li class="toctree-l1"><a class="reference internal" href="variant_summary_creation.html"><strong>Variant summary creation</strong> tool</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#"><strong>Variant recurrence table creation</strong> tool</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#about">About</a></li>
<li class="toctree-l2"><a class="reference internal" href="#input-files">Input files</a></li>
<li class="toctree-l2"><a class="reference internal" href="#running-the-tool">Running the tool</a></li>
<li class="toctree-l2"><a class="reference internal" href="#output-files">Output files</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="sample_data_postprocessing.html"><strong>Sample data post-processing</strong> tool</a></li>
<li class="toctree-l1"><a class="reference internal" href="inpred_nomenclature.html">InPreD sample ID nomenclature</a></li>
<li class="toctree-l1"><a class="reference internal" href="software_versions.html">Software versions</a></li>
<li class="toctree-l1"><a class="reference internal" href="FAQ.html">Frequently asked questions</a></li>
<li class="toctree-l1"><a class="reference internal" href="changelog.html">Change log</a></li>
</ul>

<div class="relations">
<h3>Related Topics</h3>
<ul>
  <li><a href="index.html">Documentation overview</a><ul>
      <li>Previous: <a href="variant_summary_creation.html" title="previous chapter"><strong>Variant summary creation</strong> tool</a></li>
      <li>Next: <a href="sample_data_postprocessing.html" title="next chapter"><strong>Sample data post-processing</strong> tool</a></li>
  </ul></li>
</ul>
</div>
<div id="searchbox" style="display: none" role="search">
  <h3 id="searchlabel">Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" />
      <input type="submit" value="Go" />
    </form>
    </div>
</div>
<script>$('#searchbox').show(0);</script>








        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="footer">
      &copy;2022, Daniel Vodak.
      
      |
      Powered by <a href="http://sphinx-doc.org/">Sphinx 3.4.3</a>
      &amp; <a href="https://github.com/bitprophet/alabaster">Alabaster 0.7.12</a>
      
      |
      <a href="_sources/variant_recurrence_table_creation.rst.txt"
          rel="nofollow">Page source</a>
    </div>

    

    
  </body>
</html>